bread/layout/components/datatable.py

```diff
@@ -257,15 +267,28 @@ def with_toolbar(
                 ),
                 hg.DIV(
                     hg.DIV(
-                        Search(
-                            widgetattributes={"autofocus": True},
-                            backend=search_backend,
-                            resultcontainerid=resultcontainerid,
-                            show_result_container=False,
+                        hg.FORM(
+                            searchbar,
+                            hg.Iterator(
+                                hg.C("request").GET.items(),
+                                "queryitem",
+                                hg.If(
+                                    hg.F(
+                                        lambda c: c["queryitem"][0]
+                                        != search_urlparameter
+                                    ),
+                                    hg.INPUT(
+                                        type="hidden",
+                                        name=hg.C("queryitem")[0],
+                                        value=hg.C("queryitem")[1],
+                                    ),
+                                ),
+                            ),
+                            method="GET",
```

Comment:
This is to allow the search to work without javascript. However, clearing the searchbar through the button still requires javascript.

Reply:
This is great. It looks a lot simpler than the JS one.

Reply:
RIght, I just discovered that this is a more elegant way to do it. I think there are other places in the code where we can change it to use hidden inputs to preserve the URL parameters.